src/components/Pages/Assets/Pools/AddLiquidity.tsx

```diff
@@ -0,0 +1,315 @@
+      const amount2Min = new BN(unitToPlanck(addSlippage(asset2Amount, slippage), getAssetDecimals(assetMetadata[0])));
+
+      addLiquidity(asset1, asset2, amount1, amount2, amount1Min, amount2Min);
+    },
+    [
+      api,
+      asset1,
+      asset2,
+      asset1Amount,
+      asset2Amount,
+      activeAccount,
+      assetBalance,
+      assetMetadata,
+      addLiquidity,
+      nativeBalance,
+      nativeMetadata,
+      openModalStatus,
+      poolReserves,
+      setStatus,
+    ],
+  );
+
+  const onInput1Changed = (amount1: string) => {
+    if (!nativeMetadata) return;
+    if (amount1 !== '' && !amount1.match(pricePattern(nativeMetadata.decimals))) return;
+
+    setAsset1Amount(amount1);
+    const isNewPool = isPoolEmpty(poolReserves);
+    const asset1Value = amount1;
+    const asset2Value = asset2Amount;
+
+    if (isNewPool) {
+      const exchangeRate = asset1Value && asset2Value ? calcExchangeRate(+asset1Value, +asset2Value) : null;
+      setExchangeRate(exchangeRate);
+    } else if (exchangeRate) {
+      setAsset2Amount(amount1 ? formatDecimals(exchangeRate.mul(asset1Value)) : '');
+    }
+  };
+
+  const onInput2Changed = (amount2: string) => {
+    if (!assetMetadata) return;
+    if (amount2 !== '' && !amount2.match(pricePattern(getAssetDecimals(assetMetadata[0])))) return;
+
+    setAsset2Amount(amount2);
+    const isNewPool = isPoolEmpty(poolReserves);
+    const asset1Value = asset1Amount;
+    const asset2Value = amount2;
+
+    if (isNewPool) {
+      const exchangeRate = asset1Value && asset2Value ? calcExchangeRate(+asset1Value, +asset2Value) : null;
+      setExchangeRate(exchangeRate);
+    } else if (exchangeRate) {
+      setAsset1Amount(amount2 ? formatDecimals(new Decimal(asset2Value).div(exchangeRate)) : '');
+    }
+  };
+
+  useEffect(() => {
+    if (api) {
+      if (asset1 && asset2) {
+        getAssetMetadata(asset2.asAsset).then(setAssetMetadata);
+        getPoolReserves(asset1, asset2).then(setPoolReserves);
+      }
+      if (activeAccount && asset2) {
+        getAssetBalance(asset2.asAsset).then((balance) => setAssetBalance(balance || BN_ZERO));
+        getNativeBalance();
+      }
+    }
+  }, [api, activeAccount, asset1, asset2, getAssetBalance, getAssetMetadata, getNativeBalance, getPoolReserves]);
+
+  useEffect(() => {
+    if (api && !nativeMetadata) {
+      getNativeMetadata();
+    }
+  }, [api, getNativeMetadata, nativeMetadata]);
+
+  useEffect(() => {
+    if (poolReserves && !isPoolEmpty(poolReserves) && assetMetadata && nativeMetadata) {
+      const asset1Reserve = getCleanFormattedBalance(poolReserves[0], nativeMetadata.decimals);
+      const asset2Reserve = getCleanFormattedBalance(poolReserves[1], getAssetDecimals(assetMetadata[0]));
+      setExchangeRate(calcExchangeRate(+asset1Reserve, +asset2Reserve));
+    }
+  }, [poolReserves, assetMetadata, nativeMetadata]);
```

Comment:
this is too much
should add TODO to refactor gathering data from when user starts to do stuff